test/testSetup.ts

```diff
@@ -99,11 +103,16 @@ export class TestSetup {
     await expect(uncaughtErrors.length).toBe(0);
   }
 
-  public async setUpTestingEnvironment(): Promise<void> {
+  public async setUpTestingEnvironment(repo?: constants.repoKeywords): Promise<void> {
     utilities.log('');
     utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingEnvironment()...`);
 
-    this.tempFolderPath = path.join(__dirname, '..', 'e2e-temp');
+    if (repo) {
+      const repoPath = path.join(__dirname, '..', repo);
+      await gitClone(constants.projectMaps.get(repo)!, repoPath); // clone the project
+      Env.getInstance().useExistingProject = repoPath; // set the existing project file path, value will be overriden even if USE_EXISTING_PROJECT_PATH exists
+      utilities.log(`Env.getInstance().useExistingProject - ${Env.getInstance().useExistingProject}`)
+    }
 
     this.projectFolderPath = Env.getInstance().useExistingProject
```

Comment:
Should we support the possibility that the env var could a repo URL or a directory?

Reply:
One other are that needs to be rethought is the code that cleans up a new project if present and teardown.

Let me knw your thoughts on this.